WordPress/gutenberg
# Documentation: add a more detailed guide for react native development

This PR adds some clearer first time setup instructions for react native development. I'd also be open to moving this to a separate how-to-guide, if we'd like to keep the current doc brief.

I also have images and video assets I can link to (really necessary for some of the android studio screens) but I'd be interested in knowing where the best place to add this would be. I do see an [assets](https://github.com/WordPress/gutenberg/tree/trunk/docs/assets) folder, but we probably don't want to add image/videos to the repository to avoid bloating repository size.

### Testing Instructions

- Verify that instructions are clear
- If you don't have native development going locally, give the instructions a try!
